proposals/stack-switching/Explainer.md

```diff
@@ -833,6 +833,126 @@ events and only `(on $e switch)` handlers can handle `switch`
+  - `cont ::= (E : n)`
+
+
+#### Administrative instructions
+
+* `(ref.cont a)` represents a continuation value, where `a` is a *continuation address* indexing into the store's `conts` component
+  - `ref.cont a : [] -> [(ref $ct)]`
+    - iff `S.conts[a] = epsilon \/ S.conts[a] = (E : n)`
+    - and `$ct ~~ cont $ft`
+    - and `$ft ~~ [t1^n] -> [t2*]`
+
+* `(prompt{<hdl>*} <instr>* end)` represents an active handler
+  - `(prompt{((a $l) | (b switch))*}? instr* end) : [t1*] -> [t2*]`
+    - iff `instr* : [t1*] -> [t2*]`
+    - and `(S.tags[a].type ~~ [te1*] -> [te2*])*`
+    - and `(S.tags[b].type ~~ [] -> [te2*])*`
+    - and `(label $l : [te1'* (ref null? $ct')])*`
+    - and `([te1*] <: [te1'*])*`
+    - and `($ct' ~~ cont $ft')*`
+    - and `([te2*] -> [t2*] <: $ft')*`
+
+The administrative structure `hdl` is defined as.
+```
+hdl ::= (<tagaddr> $l) hdl | (<tagaddr> switch)
+```
+
+#### Handler contexts
+
+```
+H^ea ::=
+  _
+  val* H^ea instr*
+  label_n{instr*} H^ea end
+  frame_n{F} H^ea end
+  catch{...} H^ea end
+  prompt{hdl*} H^ea end   (iff ea notin ea'*)
+```
+
+
+#### Reduction
+
+* `S; F; (ref.null t) (cont.new $ct)  -->  S; F; trap`
+
+* `S; F; (ref.func fa) (cont.new $ct)  -->  S'; F; (ref.cont |S.conts|)`
+  - iff `S' = S with conts += (E : n)`
+  - and `E = _ (invoke fa)`
+  - and `$ct ~~ cont $ft`
+  - and `$ft ~~ [t1^n] -> [t2*]`
+
+* `S; F; (ref.null t) (cont.bind $ct $ct')  -->  S; F; trap`
+
+* `S; F; (ref.cont ca) (cont.bind $ct $ct')  -->  S; F; trap`
+  - iff `S.conts[ca] = epsilon`
+
+* `S; F; v^n (ref.cont ca) (cont.bind $ct $ct')  -->  S'; F; (ref.const |S.conts|)`
+  - iff `S.conts[ca] = (E' : n')`
+  - and `$ct' ~~ cont $ft'`
+  - and `$ft' ~~ [t1'*] -> [t2'*]`
+  - and `n = n' - |t1'*|`
+  - and `S' = S with conts[ca] = epsilon with conts += (E : |t1'*|)`
+  - and `E = E'[v^n _]`
+
+* `S; F; (ref.null t) (resume $ct (on $e $l)*)  -->  S; F; trap`
+
+* `S; F; (ref.cont ca) (resume $ct (on $e $l)*)  -->  S; F; trap`
+  - iff `S.conts[ca] = epsilon`
+
+* `S; F; v^n (ref.cont ca) (resume $ct hdl*)  -->  S'; F; prompt{hdl*} E[v^n] end`
+  - iff `S.conts[ca] = (E : n)`
+  - and `(ea = F.tags[$t])*`
+  - and `S' = S with conts[ca] = epsilon`
+
+* `S; F; (ref.null t) (resume_throw $ct $e (on $t $l)*)  -->  S; F; trap`
+
+* `S; F; (ref.cont ca) (resume_throw $ct $e (on $t $l)*)  -->  S; F; trap`
+  - iff `S.conts[ca] = epsilon`
+
+* `S; F; v^m (ref.cont ca) (resume_throw $ct $e hdl*)  -->  S'; F; prompt{hdl*} E[v^m (throw $e)] end`
+  - iff `S.conts[ca] = (E : n)`
+  - and `(ea = F.tags[$t])*`
+  - and `S.tags[F.tags[$e]].type ~~ [t1^m] -> [t2*]`
+  - and `S' = S with conts[ca] = epsilon`
+
+* `S; F; (prompt{(e $l)*}? v* end)  -->  S; F; v*`
+
+* `S; F; (prompt H^ea[(suspend $e)] end)  --> S; F; trap`
+  - iff `ea = F.tags[$e]`
+
+* `S; F; (prompt{hdl1* (ea $l) hdl2*} H^ea[v^n (suspend $e)] end)  --> S'; F; v^n (ref.cont |S.conts|) (br $l)`
+  - iff `ea notin ea1*`
```

Comment:
```suggestion
  - iff `ea notin tagaddr(hdl1*)`
```

Reply:
We also need to make sure `ea` isn't handled by `H^ea`.

Reply:
Isn't that the very definition of `H^ea`?

Reply:
Maybe. Though, it isn't defined in the document currently as far as I can tell.